### PR TITLE
admin/team authorization

### PR DESCRIPTION
**Pull request by krish-andres**
Create authorization so that users can't view libraries of teams they are not a part of

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -49,4 +49,34 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
...
+    $rootScope.$on("$locationChangeStart", function(event, next, current) {
+      AuthService.currentUser(function(user) {
+        // Check if the current user is trying to visit an admin route
+        var isAdminRoute = /(admin\/)/.test(next);
+        var paramArr = next.split("/");
+        var hasPermission;
+
+        if (isAdminRoute) {
+          var libId = paramArr[6];
+          hasPermission = user.administrating.some(function(adminLib) {
```
Comment (tesk9) [inline assets/js/app.js]: I'm a little concerned about checking the team ids against the library ids. The library model actually has an attribute team that refers to the team model. Is there a reason we couldn't use that?

File: assets/js/app.js [hunk 2]
```diff
@@ -49,4 +49,34 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
...
+  app.run(["$rootScope", "$location", "AuthService", "$route", function($rootScope, $location, AuthService, $route) {
+    $rootScope.$on("$locationChangeStart", function(event, next, current) {
+      AuthService.currentUser(function(user) {
+        // Check if the current user is trying to visit an admin route
+        var isAdminRoute = /(admin\/)/.test(next);
+        var paramArr = next.split("/");
+        var hasPermission;
+
+        if (isAdminRoute) {
+          var libId = paramArr[6];
```
Comment (sdevani) [inline assets/js/app.js]: Red flag right here. You should instead use $routeParams to find the named parameter.

File: assets/js/app.js (hunk 1 above)
Comment (sdevani) [inline assets/js/app.js]: @tesk9 I believe adminLib refers to each of the libraries the admin has access to. I think the comparison is correct. @krish-andres correct me if I'm wrong

File: assets/js/app.js [hunk 3]
```diff
@@ -49,4 +49,34 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
...
+          hasPermission = user.administrating.some(function(adminLib) {
+            return adminLib.id === parseInt(libId);
+          });
+        } else {
+          var libId = paramArr[5];
```
Comment (sdevani) [inline assets/js/app.js]: Same as above. Do not hard code this number, use $routeParams

File: assets/js/app.js [hunk 4]
```diff
@@ -49,4 +49,34 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
...
+          hasPermission = user.teams.some(function(teamLib) {
```
Comment (sdevani) [inline assets/js/app.js]: does teamLib refer to a Team or a Library. If it refers to a team, you're naming this incorrectly and you're making a bad comparison.

File: assets/js/app.js (hunk 1 above)
Comment (tesk9) [inline assets/js/app.js]: The problem is that the user model does not have a direct reference to libraries.
Users have teams and teams have users.
A library has a team.
So what's going on here is that the route is giving the library id, and then the library id is being compared against the team id array found in the currentUser object.

**Comment by tesk9** [comment]
Hi! So I've fiddled with app.js a bit.

This is still not mergeable, though. This still has security issues.

There is not yet an admin check.

**Pull request closed without merging**